services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java

```diff
@@ -173,6 +186,13 @@ public LoginProtocol create(KeycloakSession session) {
 
         model = AllowedWebOriginsProtocolMapper.createClaimMapper(ALLOWED_WEB_ORIGINS);
         builtins.put(ALLOWED_WEB_ORIGINS, model);
+
+        builtins.put(IDP_REMOTE_USER_ID, UserSessionNoteMapper.createUserSessionNoteMapper(REMOTE_ID));
```

Comment:
Isn't this a mistake? The REMOTE_ID and REMOTE_USERNAME are not used as userSession notes anywhere. They are user attributes.

Reply:
Nice one, yeah. I've amended this now sorry.